packages/big-design/src/components/TableNext/Actions/Actions.tsx

```diff
@@ -50,14 +50,15 @@ const InternalActions = <T extends TableItem>({
   ...props
 }: ActionsProps<T>) => {
   const isSelectable = typeof onSelectionChange === 'function';
-  const totalItems = pagination ? pagination.totalItems : items.length;
+  const statelessPagination = pagination?.type === 'stateless';
+  const totalItems = pagination && !statelessPagination ? pagination.totalItems : items.length;
 
   const renderItemName = () => {
     if (typeof itemName !== 'string') {
       return null;
     }
 
-    const text = isSelectable ? itemName : `${totalItems} ${itemName}`;
+    const text = isSelectable || statelessPagination ? itemName : `${totalItems} ${itemName}`;
```

Comment:
I've gone back and forth on this one

`StatelessPagination` does not take a `totalItems` props for a couple of reasons

1. not all cursor pagination implementations offer a total count (it runs a little counter to the "endless feed of lots of stuff" angle)
2. if an implementation did offer a total count, we were only using it to build a range label. Users can set the range label directly by optionally setting directly `localization.rangeLabel`
3. we could add an optional `totalItems` prop and change `localization.rangeLabel` back into a function `(totalItems?: number) => string` - but then we're just giving the user back a number they just handed to us

in short, it seemed kinda redundant to have it

_however_

should a user be using a `Table` with a cursor based pagination that _does_ provide a total count, it would be nice to offer them a `x/y` summary

to achieve this, I've added an optional `totalItems` not to `StatelessPagination` itself, but to an augmented version of its props imported by `Table`: https://github.com/bigcommerce/big-design/pull/1602/commits/aba906a689200952494b2cae6a220e5741345afb

on the plus side it works and does not make `StatelessPagination`'s props less concise, the downside is that it's kinda 'secret' and not easy to document

Reply:
still on the fence, but see what you think

Reply:
I think if an API does offer `totalItems` we should provide a way to show it rather than gating it under a specific set of props. This specific part I was thinking about applies to the number of items selected, which would be valuable to show a total. It does beg the question if we should enable a consumer to localize the label though, but that can be for another PR.

Reply:
I forgot to attach the video (see below), but the idea was that https://github.com/bigcommerce/big-design/commit/aba906a689200952494b2cae6a220e5741345afb has enabled that situation, where an optional `totalItems` can be added and it'll be used in the summary


https://github.com/user-attachments/assets/030c2b4c-dc67-4724-82c5-7c9b829d7f39

the `itemName` prop can be used to localize the summary, right now we just don't hand over control of the `x/y` or if that goes before or after `itemName` - so it's probably not in high demand to localization